From 4c11faa8bb92aa435bf1796829f772d44ee33cba Mon Sep 17 00:00:00 2001
From: Kumar Gala <galak@kernel.crashing.org>
Date: Thu, 13 Nov 2014 03:21:53 -0600
Subject: [PATCH 40/53] powerpc/mpc85xx: Add fsl,qman-channel-id for the QMan
 portals

Re-add the fsl,qman-channel-id params removed by Scott from:

	commit 7f6972a0d018daacef3b31090057771bc657f18a
	Author: Kumar Gala <galak@kernel.crashing.org>
	Date:   Fri Apr 17 17:53:07 2015 -0500

	    powerpc/mpc85xx: Add FSL QorIQ DPAA QMan support to device tree(s)

	    Signed-off-by: Kumar Gala <galak@kernel.crashing.org>
	    Signed-off-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
	    Signed-off-by: Hai-Ying Wang <Haiying.Wang@freescale.com>
	    Signed-off-by: Chunhe Lan <Chunhe.Lan@freescale.com>
	    Signed-off-by: Poonam Aggrwal <poonam.aggrwal@freescale.com>
	    [Emil Medve: Sync with the upstream binding]
	    Signed-off-by: Emil Medve <Emilian.Medve@Freescale.com>
	    [Scott Wood: s/fsl,qman-channel-id/cell-index]
	    Signed-off-by: Scott Wood <scottwood@freescale.com>
	    Cc: Madalin-Cristian Bucur <madalin.bucur@freescale.com>

Signed-off-by: Madalin-Cristian Bucur <madalin.bucur@freescale.com>

Conflicts:
	arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
	arch/powerpc/boot/dts/fsl/b4si-post.dtsi
	arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
	arch/powerpc/boot/dts/fsl/t1040si-post.dtsi
	arch/powerpc/boot/dts/fsl/t2081si-post.dtsi
	arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
---
 arch/powerpc/boot/dts/fsl/b4860si-post.dtsi        | 11 +++++
 arch/powerpc/boot/dts/fsl/b4si-post.dtsi           | 14 ++++++
 arch/powerpc/boot/dts/fsl/p1023si-post.dtsi        |  3 ++
 arch/powerpc/boot/dts/fsl/qoriq-qman1-portals.dtsi | 10 +++++
 arch/powerpc/boot/dts/fsl/t1040si-post.dtsi        | 10 +++++
 arch/powerpc/boot/dts/fsl/t2081si-post.dtsi        | 18 ++++++++
 arch/powerpc/boot/dts/fsl/t4240si-post.dtsi        | 50 ++++++++++++++++++++++
 7 files changed, 116 insertions(+)

diff --git a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
index 5af90fd..ded67c1 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
@@ -173,66 +173,77 @@
 		reg = <0x38000 0x4000>, <0x100e000 0x1000>;
 		interrupts = <132 0x2 0 0>;
 		cell-index = <0xe>;
+		fsl,qman-channel-id = <0xe>;
 	};
 	qportal15: qman-portal@3c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x3c000 0x4000>, <0x100f000 0x1000>;
 		interrupts = <134 0x2 0 0>;
 		cell-index = <0xf>;
+		fsl,qman-channel-id = <0xf>;
 	};
 	qportal16: qman-portal@40000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x40000 0x4000>, <0x1010000 0x1000>;
 		interrupts = <136 0x2 0 0>;
 		cell-index = <0x10>;
+		fsl,qman-channel-id = <0x10>;
 	};
 	qportal17: qman-portal@44000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x44000 0x4000>, <0x1011000 0x1000>;
 		interrupts = <138 0x2 0 0>;
 		cell-index = <0x11>;
+		fsl,qman-channel-id = <0x11>;
 	};
 	qportal18: qman-portal@48000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x48000 0x4000>, <0x1012000 0x1000>;
 		interrupts = <140 0x2 0 0>;
 		cell-index = <0x12>;
+		fsl,qman-channel-id = <0x12>;
 	};
 	qportal19: qman-portal@4c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4c000 0x4000>, <0x1013000 0x1000>;
 		interrupts = <142 0x2 0 0>;
 		cell-index = <0x13>;
+		fsl,qman-channel-id = <0x13>;
 	};
 	qportal20: qman-portal@50000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x50000 0x4000>, <0x1014000 0x1000>;
 		interrupts = <144 0x2 0 0>;
 		cell-index = <0x14>;
+		fsl,qman-channel-id = <0x14>;
 	};
 	qportal21: qman-portal@54000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x54000 0x4000>, <0x1015000 0x1000>;
 		interrupts = <146 0x2 0 0>;
 		cell-index = <0x15>;
+		fsl,qman-channel-id = <0x15>;
 	};
 	qportal22: qman-portal@58000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x58000 0x4000>, <0x1016000 0x1000>;
 		interrupts = <148 0x2 0 0>;
 		cell-index = <0x16>;
+		fsl,qman-channel-id = <0x16>;
 	};
 	qportal23: qman-portal@5c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x5c000 0x4000>, <0x1017000 0x1000>;
 		interrupts = <150 0x2 0 0>;
 		cell-index = <0x17>;
+		fsl,qman-channel-id = <0x17>;
 	};
 	qportal24: qman-portal@60000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x60000 0x4000>, <0x1018000 0x1000>;
 		interrupts = <152 0x2 0 0>;
 		cell-index = <0x18>;
+		fsl,qman-channel-id = <0x18>;
 	};
 };
 
diff --git a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
index fafb1d9..94ff9a4 100644
--- a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
@@ -230,84 +230,98 @@
 		reg = <0x0 0x4000>, <0x1000000 0x1000>;
 		interrupts = <104 0x2 0 0>;
 		cell-index = <0x0>;
+		fsl,qman-channel-id = <0x0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x1001000 0x1000>;
 		interrupts = <106 0x2 0 0>;
 		cell-index = <0x1>;
+		fsl,qman-channel-id = <0x1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x1002000 0x1000>;
 		interrupts = <108 0x2 0 0>;
 		cell-index = <0x2>;
+		fsl,qman-channel-id = <0x2>;
 	};
 	qportal3: qman-portal@c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc000 0x4000>, <0x1003000 0x1000>;
 		interrupts = <110 0x2 0 0>;
 		cell-index = <0x3>;
+		fsl,qman-channel-id = <0x3>;
 	};
 	qportal4: qman-portal@10000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x10000 0x4000>, <0x1004000 0x1000>;
 		interrupts = <112 0x2 0 0>;
 		cell-index = <0x4>;
+		fsl,qman-channel-id = <0x4>;
 	};
 	qportal5: qman-portal@14000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x14000 0x4000>, <0x1005000 0x1000>;
 		interrupts = <114 0x2 0 0>;
 		cell-index = <0x5>;
+		fsl,qman-channel-id = <0x5>;
 	};
 	qportal6: qman-portal@18000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x18000 0x4000>, <0x1006000 0x1000>;
 		interrupts = <116 0x2 0 0>;
 		cell-index = <0x6>;
+		fsl,qman-channel-id = <0x6>;
 	};
 	qportal7: qman-portal@1c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x1c000 0x4000>, <0x1007000 0x1000>;
 		interrupts = <118 0x2 0 0>;
 		cell-index = <0x7>;
+		fsl,qman-channel-id = <0x7>;
 	};
 	qportal8: qman-portal@20000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x20000 0x4000>, <0x1008000 0x1000>;
 		interrupts = <120 0x2 0 0>;
 		cell-index = <0x8>;
+		fsl,qman-channel-id = <0x8>;
 	};
 	qportal9: qman-portal@24000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x24000 0x4000>, <0x1009000 0x1000>;
 		interrupts = <122 0x2 0 0>;
 		cell-index = <0x9>;
+		fsl,qman-channel-id = <0x9>;
 	};
 	qportal10: qman-portal@28000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x28000 0x4000>, <0x100a000 0x1000>;
 		interrupts = <124 0x2 0 0>;
 		cell-index = <0xa>;
+		fsl,qman-channel-id = <0xa>;
 	};
 	qportal11: qman-portal@2c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x2c000 0x4000>, <0x100b000 0x1000>;
 		interrupts = <126 0x2 0 0>;
 		cell-index = <0xb>;
+		fsl,qman-channel-id = <0xb>;
 	};
 	qportal12: qman-portal@30000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x30000 0x4000>, <0x100c000 0x1000>;
 		interrupts = <128 0x2 0 0>;
 		cell-index = <0xc>;
+		fsl,qman-channel-id = <0xc>;
 	};
 	qportal13: qman-portal@34000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x34000 0x4000>, <0x100d000 0x1000>;
 		interrupts = <130 0x2 0 0>;
 		cell-index = <0xd>;
+		fsl,qman-channel-id = <0xd>;
 	};
 };
 
diff --git a/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi b/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
index da6d3fc..aae0626 100644
--- a/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
@@ -122,18 +122,21 @@
 		reg = <0x0 0x4000>, <0x100000 0x1000>;
 		interrupts = <29 2 0 0>;
 		cell-index = <0>;
+		fsl,qman-channel-id = <0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x101000 0x1000>;
 		interrupts = <31 2 0 0>;
 		cell-index = <1>;
+		fsl,qman-channel-id = <1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x102000 0x1000>;
 		interrupts = <33 2 0 0>;
 		cell-index = <2>;
+		fsl,qman-channel-id = <2>;
 	};
 };
 
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-qman1-portals.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-qman1-portals.dtsi
index e77e4b4..9e60f40 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-qman1-portals.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-qman1-portals.dtsi
@@ -42,42 +42,49 @@
 		reg = <0x0 0x4000>, <0x100000 0x1000>;
 		interrupts = <104 2 0 0>;
 		cell-index = <0x0>;
+		fsl,qman-channel-id = <0x0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x101000 0x1000>;
 		interrupts = <106 2 0 0>;
 		cell-index = <1>;
+		fsl,qman-channel-id = <1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x102000 0x1000>;
 		interrupts = <108 2 0 0>;
 		cell-index = <2>;
+		fsl,qman-channel-id = <2>;
 	};
 	qportal3: qman-portal@c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc000 0x4000>, <0x103000 0x1000>;
 		interrupts = <110 2 0 0>;
 		cell-index = <3>;
+		fsl,qman-channel-id = <3>;
 	};
 	qportal4: qman-portal@10000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x10000 0x4000>, <0x104000 0x1000>;
 		interrupts = <112 2 0 0>;
 		cell-index = <4>;
+		fsl,qman-channel-id = <4>;
 	};
 	qportal5: qman-portal@14000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x14000 0x4000>, <0x105000 0x1000>;
 		interrupts = <114 2 0 0>;
 		cell-index = <5>;
+		fsl,qman-channel-id = <5>;
 	};
 	qportal6: qman-portal@18000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x18000 0x4000>, <0x106000 0x1000>;
 		interrupts = <116 2 0 0>;
 		cell-index = <6>;
+		fsl,qman-channel-id = <6>;
 	};
 
 	qportal7: qman-portal@1c000 {
@@ -85,17 +92,20 @@
 		reg = <0x1c000 0x4000>, <0x107000 0x1000>;
 		interrupts = <118 2 0 0>;
 		cell-index = <7>;
+		fsl,qman-channel-id = <7>;
 	};
 	qportal8: qman-portal@20000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x20000 0x4000>, <0x108000 0x1000>;
 		interrupts = <120 2 0 0>;
 		cell-index = <8>;
+		fsl,qman-channel-id = <8>;
 	};
 	qportal9: qman-portal@24000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x24000 0x4000>, <0x109000 0x1000>;
 		interrupts = <122 2 0 0>;
 		cell-index = <9>;
+		fsl,qman-channel-id = <9>;
 	};
 };
diff --git a/arch/powerpc/boot/dts/fsl/t1040si-post.dtsi b/arch/powerpc/boot/dts/fsl/t1040si-post.dtsi
index b738a95..6ebf311 100644
--- a/arch/powerpc/boot/dts/fsl/t1040si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t1040si-post.dtsi
@@ -300,60 +300,70 @@
 		reg = <0x0 0x4000>, <0x1000000 0x1000>;
 		interrupts = <104 0x2 0 0>;
 		cell-index = <0x0>;
+		fsl,qman-channel-id = <0x0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x1001000 0x1000>;
 		interrupts = <106 0x2 0 0>;
 		cell-index = <0x1>;
+		fsl,qman-channel-id = <0x1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x1002000 0x1000>;
 		interrupts = <108 0x2 0 0>;
 		cell-index = <0x2>;
+		fsl,qman-channel-id = <0x2>;
 	};
 	qportal3: qman-portal@c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc000 0x4000>, <0x1003000 0x1000>;
 		interrupts = <110 0x2 0 0>;
 		cell-index = <0x3>;
+		fsl,qman-channel-id = <0x3>;
 	};
 	qportal4: qman-portal@10000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x10000 0x4000>, <0x1004000 0x1000>;
 		interrupts = <112 0x2 0 0>;
 		cell-index = <0x4>;
+		fsl,qman-channel-id = <0x4>;
 	};
 	qportal5: qman-portal@14000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x14000 0x4000>, <0x1005000 0x1000>;
 		interrupts = <114 0x2 0 0>;
 		cell-index = <0x5>;
+		fsl,qman-channel-id = <0x5>;
 	};
 	qportal6: qman-portal@18000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x18000 0x4000>, <0x1006000 0x1000>;
 		interrupts = <116 0x2 0 0>;
 		cell-index = <0x6>;
+		fsl,qman-channel-id = <0x6>;
 	};
 	qportal7: qman-portal@1c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x1c000 0x4000>, <0x1007000 0x1000>;
 		interrupts = <118 0x2 0 0>;
 		cell-index = <0x7>;
+		fsl,qman-channel-id = <0x7>;
 	};
 	qportal8: qman-portal@20000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x20000 0x4000>, <0x1008000 0x1000>;
 		interrupts = <120 0x2 0 0>;
 		cell-index = <0x8>;
+		fsl,qman-channel-id = <0x8>;
 	};
 	qportal9: qman-portal@24000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x24000 0x4000>, <0x1009000 0x1000>;
 		interrupts = <122 0x2 0 0>;
 		cell-index = <0x9>;
+		fsl,qman-channel-id = <0x9>;
 	};
 };
 
diff --git a/arch/powerpc/boot/dts/fsl/t2081si-post.dtsi b/arch/powerpc/boot/dts/fsl/t2081si-post.dtsi
index f836bc3..8bcdb35 100644
--- a/arch/powerpc/boot/dts/fsl/t2081si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t2081si-post.dtsi
@@ -346,108 +346,126 @@
 		reg = <0x0 0x4000>, <0x1000000 0x1000>;
 		interrupts = <104 0x2 0 0>;
 		cell-index = <0x0>;
+		fsl,qman-channel-id = <0x0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x1001000 0x1000>;
 		interrupts = <106 0x2 0 0>;
 		cell-index = <0x1>;
+		fsl,qman-channel-id = <0x1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x1002000 0x1000>;
 		interrupts = <108 0x2 0 0>;
 		cell-index = <0x2>;
+		fsl,qman-channel-id = <0x2>;
 	};
 	qportal3: qman-portal@c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc000 0x4000>, <0x1003000 0x1000>;
 		interrupts = <110 0x2 0 0>;
 		cell-index = <0x3>;
+		fsl,qman-channel-id = <0x3>;
 	};
 	qportal4: qman-portal@10000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x10000 0x4000>, <0x1004000 0x1000>;
 		interrupts = <112 0x2 0 0>;
 		cell-index = <0x4>;
+		fsl,qman-channel-id = <0x4>;
 	};
 	qportal5: qman-portal@14000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x14000 0x4000>, <0x1005000 0x1000>;
 		interrupts = <114 0x2 0 0>;
 		cell-index = <0x5>;
+		fsl,qman-channel-id = <0x5>;
 	};
 	qportal6: qman-portal@18000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x18000 0x4000>, <0x1006000 0x1000>;
 		interrupts = <116 0x2 0 0>;
 		cell-index = <0x6>;
+		fsl,qman-channel-id = <0x6>;
 	};
 	qportal7: qman-portal@1c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x1c000 0x4000>, <0x1007000 0x1000>;
 		interrupts = <118 0x2 0 0>;
 		cell-index = <0x7>;
+		fsl,qman-channel-id = <0x7>;
 	};
 	qportal8: qman-portal@20000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x20000 0x4000>, <0x1008000 0x1000>;
 		interrupts = <120 0x2 0 0>;
 		cell-index = <0x8>;
+		fsl,qman-channel-id = <0x8>;
 	};
 	qportal9: qman-portal@24000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x24000 0x4000>, <0x1009000 0x1000>;
 		interrupts = <122 0x2 0 0>;
 		cell-index = <0x9>;
+		fsl,qman-channel-id = <0x9>;
 	};
 	qportal10: qman-portal@28000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x28000 0x4000>, <0x100a000 0x1000>;
 		interrupts = <124 0x2 0 0>;
 		cell-index = <0xa>;
+		fsl,qman-channel-id = <0xa>;
 	};
 	qportal11: qman-portal@2c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x2c000 0x4000>, <0x100b000 0x1000>;
 		interrupts = <126 0x2 0 0>;
 		cell-index = <0xb>;
+		fsl,qman-channel-id = <0xb>;
 	};
 	qportal12: qman-portal@30000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x30000 0x4000>, <0x100c000 0x1000>;
 		interrupts = <128 0x2 0 0>;
 		cell-index = <0xc>;
+		fsl,qman-channel-id = <0xc>;
 	};
 	qportal13: qman-portal@34000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x34000 0x4000>, <0x100d000 0x1000>;
 		interrupts = <130 0x2 0 0>;
 		cell-index = <0xd>;
+		fsl,qman-channel-id = <0xd>;
 	};
 	qportal14: qman-portal@38000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x38000 0x4000>, <0x100e000 0x1000>;
 		interrupts = <132 0x2 0 0>;
 		cell-index = <0xe>;
+		fsl,qman-channel-id = <0xe>;
 	};
 	qportal15: qman-portal@3c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x3c000 0x4000>, <0x100f000 0x1000>;
 		interrupts = <134 0x2 0 0>;
 		cell-index = <0xf>;
+		fsl,qman-channel-id = <0xf>;
 	};
 	qportal16: qman-portal@40000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x40000 0x4000>, <0x1010000 0x1000>;
 		interrupts = <136 0x2 0 0>;
 		cell-index = <0x10>;
+		fsl,qman-channel-id = <0x10>;
 	};
 	qportal17: qman-portal@44000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x44000 0x4000>, <0x1011000 0x1000>;
 		interrupts = <138 0x2 0 0>;
 		cell-index = <0x11>;
+		fsl,qman-channel-id = <0x11>;
 	};
 };
 
diff --git a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
index c6a51ad..a4b4a23 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
@@ -576,300 +576,350 @@
 		reg = <0x0 0x4000>, <0x1000000 0x1000>;
 		interrupts = <104 0x2 0 0>;
 		cell-index = <0x0>;
+		fsl,qman-channel-id = <0x0>;
 	};
 	qportal1: qman-portal@4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4000 0x4000>, <0x1001000 0x1000>;
 		interrupts = <106 0x2 0 0>;
 		cell-index = <0x1>;
+		fsl,qman-channel-id = <0x1>;
 	};
 	qportal2: qman-portal@8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8000 0x4000>, <0x1002000 0x1000>;
 		interrupts = <108 0x2 0 0>;
 		cell-index = <0x2>;
+		fsl,qman-channel-id = <0x2>;
 	};
 	qportal3: qman-portal@c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc000 0x4000>, <0x1003000 0x1000>;
 		interrupts = <110 0x2 0 0>;
 		cell-index = <0x3>;
+		fsl,qman-channel-id = <0x3>;
 	};
 	qportal4: qman-portal@10000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x10000 0x4000>, <0x1004000 0x1000>;
 		interrupts = <112 0x2 0 0>;
 		cell-index = <0x4>;
+		fsl,qman-channel-id = <0x4>;
 	};
 	qportal5: qman-portal@14000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x14000 0x4000>, <0x1005000 0x1000>;
 		interrupts = <114 0x2 0 0>;
 		cell-index = <0x5>;
+		fsl,qman-channel-id = <0x5>;
 	};
 	qportal6: qman-portal@18000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x18000 0x4000>, <0x1006000 0x1000>;
 		interrupts = <116 0x2 0 0>;
 		cell-index = <0x6>;
+		fsl,qman-channel-id = <0x6>;
 	};
 	qportal7: qman-portal@1c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x1c000 0x4000>, <0x1007000 0x1000>;
 		interrupts = <118 0x2 0 0>;
 		cell-index = <0x7>;
+		fsl,qman-channel-id = <0x7>;
 	};
 	qportal8: qman-portal@20000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x20000 0x4000>, <0x1008000 0x1000>;
 		interrupts = <120 0x2 0 0>;
 		cell-index = <0x8>;
+		fsl,qman-channel-id = <0x8>;
 	};
 	qportal9: qman-portal@24000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x24000 0x4000>, <0x1009000 0x1000>;
 		interrupts = <122 0x2 0 0>;
 		cell-index = <0x9>;
+		fsl,qman-channel-id = <0x9>;
 	};
 	qportal10: qman-portal@28000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x28000 0x4000>, <0x100a000 0x1000>;
 		interrupts = <124 0x2 0 0>;
 		cell-index = <0xa>;
+		fsl,qman-channel-id = <0xa>;
 	};
 	qportal11: qman-portal@2c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x2c000 0x4000>, <0x100b000 0x1000>;
 		interrupts = <126 0x2 0 0>;
 		cell-index = <0xb>;
+		fsl,qman-channel-id = <0xb>;
 	};
 	qportal12: qman-portal@30000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x30000 0x4000>, <0x100c000 0x1000>;
 		interrupts = <128 0x2 0 0>;
 		cell-index = <0xc>;
+		fsl,qman-channel-id = <0xc>;
 	};
 	qportal13: qman-portal@34000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x34000 0x4000>, <0x100d000 0x1000>;
 		interrupts = <130 0x2 0 0>;
 		cell-index = <0xd>;
+		fsl,qman-channel-id = <0xd>;
 	};
 	qportal14: qman-portal@38000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x38000 0x4000>, <0x100e000 0x1000>;
 		interrupts = <132 0x2 0 0>;
 		cell-index = <0xe>;
+		fsl,qman-channel-id = <0xe>;
 	};
 	qportal15: qman-portal@3c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x3c000 0x4000>, <0x100f000 0x1000>;
 		interrupts = <134 0x2 0 0>;
 		cell-index = <0xf>;
+		fsl,qman-channel-id = <0xf>;
 	};
 	qportal16: qman-portal@40000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x40000 0x4000>, <0x1010000 0x1000>;
 		interrupts = <136 0x2 0 0>;
 		cell-index = <0x10>;
+		fsl,qman-channel-id = <0x10>;
 	};
 	qportal17: qman-portal@44000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x44000 0x4000>, <0x1011000 0x1000>;
 		interrupts = <138 0x2 0 0>;
 		cell-index = <0x11>;
+		fsl,qman-channel-id = <0x11>;
 	};
 	qportal18: qman-portal@48000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x48000 0x4000>, <0x1012000 0x1000>;
 		interrupts = <140 0x2 0 0>;
 		cell-index = <0x12>;
+		fsl,qman-channel-id = <0x12>;
 	};
 	qportal19: qman-portal@4c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x4c000 0x4000>, <0x1013000 0x1000>;
 		interrupts = <142 0x2 0 0>;
 		cell-index = <0x13>;
+		fsl,qman-channel-id = <0x13>;
 	};
 	qportal20: qman-portal@50000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x50000 0x4000>, <0x1014000 0x1000>;
 		interrupts = <144 0x2 0 0>;
 		cell-index = <0x14>;
+		fsl,qman-channel-id = <0x14>;
 	};
 	qportal21: qman-portal@54000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x54000 0x4000>, <0x1015000 0x1000>;
 		interrupts = <146 0x2 0 0>;
 		cell-index = <0x15>;
+		fsl,qman-channel-id = <0x15>;
 	};
 	qportal22: qman-portal@58000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x58000 0x4000>, <0x1016000 0x1000>;
 		interrupts = <148 0x2 0 0>;
 		cell-index = <0x16>;
+		fsl,qman-channel-id = <0x16>;
 	};
 	qportal23: qman-portal@5c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x5c000 0x4000>, <0x1017000 0x1000>;
 		interrupts = <150 0x2 0 0>;
 		cell-index = <0x17>;
+		fsl,qman-channel-id = <0x17>;
 	};
 	qportal24: qman-portal@60000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x60000 0x4000>, <0x1018000 0x1000>;
 		interrupts = <152 0x2 0 0>;
 		cell-index = <0x18>;
+		fsl,qman-channel-id = <0x18>;
 	};
 	qportal25: qman-portal@64000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x64000 0x4000>, <0x1019000 0x1000>;
 		interrupts = <154 0x2 0 0>;
 		cell-index = <0x19>;
+		fsl,qman-channel-id = <0x19>;
 	};
 	qportal26: qman-portal@68000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x68000 0x4000>, <0x101a000 0x1000>;
 		interrupts = <156 0x2 0 0>;
 		cell-index = <0x1a>;
+		fsl,qman-channel-id = <0x1a>;
 	};
 	qportal27: qman-portal@6c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x6c000 0x4000>, <0x101b000 0x1000>;
 		interrupts = <158 0x2 0 0>;
 		cell-index = <0x1b>;
+		fsl,qman-channel-id = <0x1b>;
 	};
 	qportal28: qman-portal@70000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x70000 0x4000>, <0x101c000 0x1000>;
 		interrupts = <160 0x2 0 0>;
 		cell-index = <0x1c>;
+		fsl,qman-channel-id = <0x1c>;
 	};
 	qportal29: qman-portal@74000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x74000 0x4000>, <0x101d000 0x1000>;
 		interrupts = <162 0x2 0 0>;
 		cell-index = <0x1d>;
+		fsl,qman-channel-id = <0x1d>;
 	};
 	qportal30: qman-portal@78000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x78000 0x4000>, <0x101e000 0x1000>;
 		interrupts = <164 0x2 0 0>;
 		cell-index = <0x1e>;
+		fsl,qman-channel-id = <0x1e>;
 	};
 	qportal31: qman-portal@7c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x7c000 0x4000>, <0x101f000 0x1000>;
 		interrupts = <166 0x2 0 0>;
 		cell-index = <0x1f>;
+		fsl,qman-channel-id = <0x1f>;
 	};
 	qportal32: qman-portal@80000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x80000 0x4000>, <0x1020000 0x1000>;
 		interrupts = <168 0x2 0 0>;
 		cell-index = <0x20>;
+		fsl,qman-channel-id = <0x20>;
 	};
 	qportal33: qman-portal@84000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x84000 0x4000>, <0x1021000 0x1000>;
 		interrupts = <170 0x2 0 0>;
 		cell-index = <0x21>;
+		fsl,qman-channel-id = <0x21>;
 	};
 	qportal34: qman-portal@88000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x88000 0x4000>, <0x1022000 0x1000>;
 		interrupts = <172 0x2 0 0>;
 		cell-index = <0x22>;
+		fsl,qman-channel-id = <0x22>;
 	};
 	qportal35: qman-portal@8c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x8c000 0x4000>, <0x1023000 0x1000>;
 		interrupts = <174 0x2 0 0>;
 		cell-index = <0x23>;
+		fsl,qman-channel-id = <0x23>;
 	};
 	qportal36: qman-portal@90000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x90000 0x4000>, <0x1024000 0x1000>;
 		interrupts = <384 0x2 0 0>;
 		cell-index = <0x24>;
+		fsl,qman-channel-id = <0x24>;
 	};
 	qportal37: qman-portal@94000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x94000 0x4000>, <0x1025000 0x1000>;
 		interrupts = <386 0x2 0 0>;
 		cell-index = <0x25>;
+		fsl,qman-channel-id = <0x25>;
 	};
 	qportal38: qman-portal@98000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x98000 0x4000>, <0x1026000 0x1000>;
 		interrupts = <388 0x2 0 0>;
 		cell-index = <0x26>;
+		fsl,qman-channel-id = <0x26>;
 	};
 	qportal39: qman-portal@9c000 {
 		compatible = "fsl,qman-portal";
 		reg = <0x9c000 0x4000>, <0x1027000 0x1000>;
 		interrupts = <390 0x2 0 0>;
 		cell-index = <0x27>;
+		fsl,qman-channel-id = <0x27>;
 	};
 	qportal40: qman-portal@a0000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xa0000 0x4000>, <0x1028000 0x1000>;
 		interrupts = <392 0x2 0 0>;
 		cell-index = <0x28>;
+		fsl,qman-channel-id = <0x28>;
 	};
 	qportal41: qman-portal@a4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xa4000 0x4000>, <0x1029000 0x1000>;
 		interrupts = <394 0x2 0 0>;
 		cell-index = <0x29>;
+		fsl,qman-channel-id = <0x29>;
 	};
 	qportal42: qman-portal@a8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xa8000 0x4000>, <0x102a000 0x1000>;
 		interrupts = <396 0x2 0 0>;
 		cell-index = <0x2a>;
+		fsl,qman-channel-id = <0x2a>;
 	};
 	qportal43: qman-portal@ac000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xac000 0x4000>, <0x102b000 0x1000>;
 		interrupts = <398 0x2 0 0>;
 		cell-index = <0x2b>;
+		fsl,qman-channel-id = <0x2b>;
 	};
 	qportal44: qman-portal@b0000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xb0000 0x4000>, <0x102c000 0x1000>;
 		interrupts = <400 0x2 0 0>;
 		cell-index = <0x2c>;
+		fsl,qman-channel-id = <0x2c>;
 	};
 	qportal45: qman-portal@b4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xb4000 0x4000>, <0x102d000 0x1000>;
 		interrupts = <402 0x2 0 0>;
 		cell-index = <0x2d>;
+		fsl,qman-channel-id = <0x2d>;
 	};
 	qportal46: qman-portal@b8000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xb8000 0x4000>, <0x102e000 0x1000>;
 		interrupts = <404 0x2 0 0>;
 		cell-index = <0x2e>;
+		fsl,qman-channel-id = <0x2e>;
 	};
 	qportal47: qman-portal@bc000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xbc000 0x4000>, <0x102f000 0x1000>;
 		interrupts = <406 0x2 0 0>;
 		cell-index = <0x2f>;
+		fsl,qman-channel-id = <0x2f>;
 	};
 	qportal48: qman-portal@c0000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc0000 0x4000>, <0x1030000 0x1000>;
 		interrupts = <408 0x2 0 0>;
 		cell-index = <0x30>;
+		fsl,qman-channel-id = <0x30>;
 	};
 	qportal49: qman-portal@c4000 {
 		compatible = "fsl,qman-portal";
 		reg = <0xc4000 0x4000>, <0x1031000 0x1000>;
 		interrupts = <410 0x2 0 0>;
 		cell-index = <0x31>;
+		fsl,qman-channel-id = <0x31>;
 	};
 };
 
-- 
2.1.0.27.g96db324

